src/ppx_import.ml

```diff
@@ -465,79 +465,72 @@ let rec psig_of_tsig ~subst (tsig : Compat.signature_item_407 list) :
-                Location.raise_errorf ~loc
-                  "[%%import] cannot import a functor application %s"
-                  (string_of_lid lid)
-              | Longident.Lident _ as head_id ->
-                (* In this case, we know for sure that the user intends this lident
-                   as a module type name, so we use Typetexp.find_type and
-                   let the failure cases propagate to the user. *)
-                Compat.find_modtype env ~loc head_id |> snd
-              | Longident.Ldot (parent_id, elem) ->
-                let sig_items = locate_sig ~loc env parent_id in
-                get_modtype_decl ~loc sig_items parent_id elem
-            in
-            match tmodtype_decl with
-            | {mtd_type = Some (Mty_signature tsig); _} ->
-              let subst =
-                List.map (fun ({txt; _}, typ) -> (`Lid txt, typ)) subst
-              in
-              let psig =
-                psig_of_tsig ~subst
-                  (List.map Compat.migrate_signature_item tsig)
-              in
-              {modtype_decl with pmty_desc = Pmty_signature psig}
-            | {mtd_type = None; _} ->
-              Location.raise_errorf ~loc "Imported module is abstract"
-            | _ ->
-              Location.raise_errorf ~loc "Imported module is indirectly defined")
-    | _ -> Location.raise_errorf ~loc "Invalid [%%import] syntax" )
-  | _ -> modtype_decl
+  let ({txt = lid; loc} as alias), subst = package_type in
+  if tool_name = "ocamldep" then
+    if is_self_reference ~input_name lid then
+      (* Create a dummy module type to break the circular dependency *)
+      Ast_helper.Mty.mk ~attrs:[] (Pmty_signature [])
+    else
+      (* Just put it as alias *)
+      Ast_helper.Mty.mk ~attrs:[] (Pmty_alias alias)
+  else
+    Ppxlib.Ast_helper.with_default_loc loc (fun () ->
+        let env = Lazy.force lazy_env in
+        let tmodtype_decl =
+          match lid with
+          | Longident.Lapply _ ->
+            Location.raise_errorf ~loc
+              "[%%import] cannot import a functor application %s"
+              (string_of_lid lid)
+          | Longident.Lident _ as head_id ->
+            (* In this case, we know for sure that the user intends this lident
+               as a module type name, so we use Typetexp.find_type and
+               let the failure cases propagate to the user. *)
+            Compat.find_modtype env ~loc head_id |> snd
+          | Longident.Ldot (parent_id, elem) ->
+            let sig_items = locate_sig ~loc env parent_id in
+            get_modtype_decl ~loc sig_items parent_id elem
+        in
+        match tmodtype_decl with
+        | {mtd_type = Some (Mty_signature tsig); _} ->
+          let subst = List.map (fun ({txt; _}, typ) -> (`Lid txt, typ)) subst in
+          let psig =
+            psig_of_tsig ~subst (List.map Compat.migrate_signature_item tsig)
+          in
+          Ast_helper.Mty.mk ~attrs:[] (Pmty_signature psig)
```

Comment:
> Since we no longer receive a `module_type` as input, I had now to construct one from scratch. Not sure how it matters, especially since all the tests are passing (including the one that was failing before).

This place is what you were referring to, isn't it?

I'm not sure if that ever might happen, but if someone wrote 
```
module type Foo = [%import: ...] [@inner_attribute] [@@deriving ...]
```
then the inner attribute would get lost now. 

@ejgallego, do you think that's a problem? By the way, I'm not really sure: what's the use case of `ppx_import` in the context of module types? Is it really also for derivers (as in my example above) or is it actually something different? Or both?

Reply:
I can't tell, I'll test when I can, but I think it should be OK for my use cases.

Reply:
Thanks! Fingers crossed.

Reply:
So you have use cases of `ppx_import` on module types yourself?

Reply:
@ejgallego, how is this going? Have you already had time to test this?

Reply:
Nope sorry, but will go ahead ASAP.

Reply:
> This place is what you were referring to, isn't it?

Yes indeed. And also [here](https://github.com/ocaml-ppx/ppx_import/pull/54/commits/23fd6a8b8ff7c7f2dccad08dcc013e4c3501de6b#diff-2643fbce85cdaa736b4e63b3d83b2ba311bf7eb40d0ba9775a6a11f0198f0854R475) and [here](https://github.com/ocaml-ppx/ppx_import/pull/54/commits/23fd6a8b8ff7c7f2dccad08dcc013e4c3501de6b#diff-2643fbce85cdaa736b4e63b3d83b2ba311bf7eb40d0ba9775a6a11f0198f0854R478) but as you pointed out, these are just for `ocamldep` so that's probably fine ?

Reply:
@ejgallego, kind reminder. Have you already had the chance to look into this? :)

Reply:
@pitag-ha @tatchi , sorry for the delay, I'm happy to report that pinning `ppx_import` to this PR and using OCaml 4.12.1, coq-serapi test suite seems work fine and passes the test suite! All without a single change to the source code, congrats on the amazing work!